packages/sdk/src/types/runtime/appearance.ts

```diff
@@ -128,16 +128,16 @@ export class Appearance implements AppearanceLike {
 
 	/** @returns A shared reference to this actor's mesh, or null if this actor has no mesh */
 	public get mesh() {
-		return this.actor.context.internal.lookupAsset(this._meshId) as Mesh;
+		return this.actor.context.internal.lookupAsset(this._meshId).mesh;
 	}
 	public set mesh(value) {
-		this.meshId = value && value.id || ZeroGuid;
+		this.meshId = value?.id ?? ZeroGuid;
 	}
 
 	/** @inheritdoc */
 	public get meshId() { return this._meshId; }
 	public set meshId(value) {
-		if (!value || value.startsWith('0000')) {
+		if (!value) {
```

Comment:
>value [](start = 7, length = 5)

The logic here used to also check for a non-null value of all zeros. How is that accounted for now?

Reply:
All-zero GUIDs don't need this path, since the "then" part is to use the Zero guid instead.

---
In reply to: [372083083](https://github.com/microsoft/mixed-reality-extension-sdk/pull/476#discussion_r372083083) [](ancestors = 372083083)